lib/awful/ewmh.lua

```diff
@@ -246,6 +246,12 @@ local context_mapper = {
 function ewmh.geometry(c, context, hints)
     local layout = c.screen.selected_tag and c.screen.selected_tag.layout or nil
 
+    -- For dockable clients, always apply new geometry
+    if c.dockable and context == "ewmh" then
+        c:geometry(hints.geometry)
+        return
+    end
+
```

Comment:
@Elv13 That will work for preserving behaviour, but ideally you would want to handle requests according to the layout. Bug #1554 will persist if this approach is taken.